modules/mkdocs.py

```diff
@@ -0,0 +1,165 @@
+import asyncio
+from modules.common.module import BotModule
+import requests
+import re
+
+class MatrixModule(BotModule):
+    def __init__(self, name):
+        super().__init__(name)
+        self.rooms = dict()
+        self.sites = dict()
+
+    def set_settings(self, data):
+        super().set_settings(data)
+        if data.get('rooms'):
+            self.search_index = data['rooms']
+        if data.get('sites'):
+            self.shows = data['sites']
+
+    def get_settings(self):
+        data = super().get_settings()
+        data['rooms'] = self.rooms
+        data['sites'] = self.shows
+        return data
+
+
+    def matrix_start(self, bot):
+        super().matrix_start(bot)
+        self.bot = bot
+        self.add_module_aliases(bot, ['search','find'])
+
+    async def matrix_message(self, bot, room, event):
+        args = event.body.split()
+        cmd = args.pop(0).lower()
+        if cmd == f'!{self.name}' and len(args):
+            cmd = args.pop(0).lower()
+        if cmd[0] == '!':
+            cmd = cmd[1:]
+        if cmd in ['setup']:
+            bot.must_be_admin(room, event)
+            self.logger.info(f"Setting up search for room {room.name}")
+            self.rooms[room.room_id] = {
```

Comment:
What's the rationale for a unique site/list of sites per room?

Reply:
The idea is that each room could have it's own `notes` site. So the self-hosted room could have !search notes point to a separate site from the main JB rooms, which have !search notes point to jupiterbroadcasting notes site.

As I type this out, I guess that would just mean that the aliases are room unique.